jdaviz/configs/default/plugins/metadata_viewer/metadata_viewer.py

```diff
@@ -0,0 +1,111 @@
+from astropy.io.fits import Header
+from traitlets import Bool, List, Unicode, observe
+
+from jdaviz.core.events import AddDataMessage, RemoveDataMessage
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import TemplateMixin
+
+__all__ = ['MetadataViewer']
+
+
+@tray_registry('g-metadata-viewer', label="Metadata Viewer")
+class MetadataViewer(TemplateMixin):
+    template_file = __file__, "metadata_viewer.vue"
+    dc_items = List([]).tag(sync=True)
+    selected_data = Unicode("").tag(sync=True)
+    has_metadata = Bool(False).tag(sync=True)
+    metadata = List([]).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.hub.subscribe(self, AddDataMessage,
+                           handler=self._on_viewer_data_changed)
+        self.hub.subscribe(self, RemoveDataMessage,
+                           handler=self._on_viewer_data_changed)
+
+    def _on_viewer_data_changed(self, *args, **kwargs):
+        self.dc_items = [data.label for data in self.app.data_collection]
+
+    @observe("selected_data")
+    def _show_metadata(self, event):
+        try:
+            data = self.app.data_collection[self.app.data_collection.labels.index(event['new'])]
+        except ValueError:
+            self.has_metadata = False
+            self.metadata = []
+            return
+
+        if not hasattr(data, 'meta') or not isinstance(data.meta, dict) or len(data.meta) < 1:
+            self.has_metadata = False
+            self.metadata = []
+        else:
+            if 'header' in data.meta and isinstance(data.meta['header'], (dict, Header)):
+                if isinstance(data.meta['header'], Header):  # Specviz
+                    meta = dict(data.meta['header'])
+                else:
+                    meta = data.meta['header']
+            else:
+                meta = data.meta
+
+            d = flatten_nested_dict(meta)
+            for badkey in ('COMMENT', 'HISTORY', ''):

```

Comment:
Is this general or just jwst-specific (i.e. can we check if the content is json-able across all entries rather than checking for these specific keys)?  And either way - should this be mentioned in the docs or in a snackbar message (otherwise a user might be confused if they were expecting more entries to show)?

Reply:
Ok, but would it be too expensive to loop over the whole dictionary and exclude any cases that fail to be cleaned to avoid failing for other cases we can't predict (possibly by just trying `json.dumps(value)`)?  If that isn't feasible, then this is probably fine for now for JWST/HST.

Reply:
I don't think performance is an issue here, all things considered. But I wonder if it might be misleading to automatically exclude other stuff, which might be real bugs we might want to fix. At least, I know that no one sane is going to want to see all the comments and history lines (some might but...).